File: java/arcade/IsIPv4Address.java
import java.util.regex.*;

public final class IsIPv4Address {

	boolean isIPv4Address(String inputString) {
		String reg = "\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}";
		Matcher mt = Pattern.compile(reg).matcher(inputString);
		int c = 0;
		if(mt.matches()) {
			for(String s : inputString.split("\\.")) {
				c += (Integer.valueOf(s) > 255) ? 1 : 0;
			}
		}
		return (mt.matches() && c == 0);
	}
}

File: ruby/arcade/isIPv4Address.rb
def isIPv4Address(inputString)
	reg, c = /\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}/, 0
    b = reg.match(inputString).to_s.eql?(inputString)
    inputString.split("\.").each do |i|
        c += i.to_i > 255 ? 1 : 0
    end if b
    return (b and c == 0)
end